### PR TITLE
Handle RTL for the Modal Component

### LINKED ISSUE #2756: RTL - Modal Component
### Expected Behavior

Libraries and modals that use the Modal component should work in RTL languages (e.g. Hebrew, Arabic)

Splitting out from #1579

### PR DESCRIPTION
**Pull request by chrisgarrity**
### Resolves

Fixes #2756 

### Proposed Changes

Libraries and many other modals share the Modal Component.

Connected the Modal container to redux to make it aware of the RTL state, and then made sure that all places that use the modal component import the container, not the component directly.

### Test Coverage

* Current tests still run
Testing: [Add `/?locale=he` to the URL]
- [ ] All libraries open with the 'back' button on the right (sprites, costumes, backdrops, sounds, tutorials), and the list reversed
- [ ] new/edit variable modal should be RTL
- [ ] new/edit list modal should be RTL
- [ ] custom procedure modal should be RTL
- [ ] camera modal should be RTL
- [ ] hardware extension connection modals should be RTL and work
  - [ ] micro:bit
  - [ ] EV3

### Browser Coverage
Check the OS/browser combinations tested (At least 2)

Mac
 * [ ] Chrome 
 * [ ] Firefox 
 * [ ] Safari
 
Windows
 * [ ] Chrome 
 * [ ] Firefox 
 * [ ] Edge
 
Chromebook
 * [ ] Chrome
 
iPad
* [ ] Safari

Android Tablet
* [ ] Chrome

### CODE REVIEW

File: src/components/connection-modal/connection-modal.css [hunk 1]
```diff
@@ -51,9 +51,12 @@
     align-items: center;
 }
 
-.device-tile-image {
+[dir="ltr"].device-tile-image {
```
Comment (kchadha) [inline src/components/connection-modal/connection-modal.css]: Nit: here, you don't have a space between `[dir=:"ltr"]` and the `.<class-name>` where in the other changes below, you do have a space. The style should be consistent.

**Comment by kchadha** [comment]
Things I Noticed While Testing:

- The library title is no longer centered in RTL or LTR
- Record Modal/Camera Modal - the 'Re-record' and 'Retake' buttons should have the arrow icon flipped in RTL:
image

- The block in the custom procedure modal should also display in RTL (I think this is an option we can pass in to the custom procedure modal's block workspace in the custom-procedures.jsx container).

image

- Missing some space between icons and corresponding text, examples:
    - the re-record button above (retake in the camera modal looks okay though)
    - image vs. image
    - library back button

**Comment by kchadha** [comment]
Two more things I noticed:

- Custom procedure modal -- adding new inputs etc. does not re-center the block correctly.
- The extension modal help button is right up against the edge of the modal instead of being in-set a little bit like the modal close button:
image

**Comment by chrisgarrity** [comment]
@kchadha one last check - it's built on github: https://chrisgarrity.github.io/scratch-gui/feature/2756-rtl-libraries/?locale=he
This is everything except the custom procedures.

**Comment by kchadha** [comment]
I just noticed that the ? icon gets flipped in one of the buttons on the extensions modal

image

**Comment by chrisgarrity** [comment]
@kchadha changed the connection modal to only flip the back icon direction on the error and unavailable modals. It's currently building, so should be available on my github soon.

**Comment by chrisgarrity** [comment]
@BryceLTaylor assigned you as this should be in the smoke-test.

**Pull request merged**